Framework/HistogramData/src/Histogram.cpp

```diff
@@ -140,6 +159,9 @@ void Histogram::setSharedX(const Kernel::cow_ptr<HistogramX> &x) & {
 
   Throws if the size does not match the current size. */
 void Histogram::setSharedY(const Kernel::cow_ptr<HistogramY> &y) & {
+  if (yMode() == YMode::Uninitialized)
+    throw std::logic_error(
+        "Histogram::setSharedY: YMode is not set and cannot be determined");
```

Comment:
It's more logically associated with the `HistogramY`, so would probably best live there. However, I don't think it matters too much at this stage. I guess the conversion code counts->frequencies, frequencies->counts still makes most sense on `Histogram` as it touches all three Histogram vectors.